pyretailscience/plots/line.py

```diff
@@ -0,0 +1,317 @@
+    category_totals = df_since_competitor_opened.groupby('category_0_name')['total_price'].sum().sort_values(ascending=False)
+
+    # Get the top 3 categories
+    top_categories = category_totals.head(3).index
+
+    # Filter the dataframe to include only the top 3 categories
+    df_top_categories = df_since_competitor_opened[df_since_competitor_opened['category_0_name'].isin(top_categories)]
+    ```
+
+    **Plotting**:
+    ```python
+    ax = line.plot(
+        df=df_top_categories,
+        value_col="total_price",  # Plot 'total_price' values
+        group_col="category_0_name",  # Separate lines for each category
+        x_col="months_since_competitor_opened",  # Use 'months_since_competitor_opened' as the x-axis
+        title="Total Price for Top 3 Categories (Before and After Competitor Opened)",  # Title of the plot
+        x_label="Months Since Competitor Opened",  # X-axis label
+        y_label="Total Price",  # Y-axis label
+        legend_title="Category"  # Legend title
+    )
+
+    plt.show()
+    ```
+
+    **Use Case**: Use this when you want to analyze the behavior of specific top categories before and after
+    an event, such as the opening of a competitor store.
+
+---
+
+### Customization Options:
+- **`value_col`**: The column or list of columns to plot (e.g., `'total_price'`).
+- **`group_col`**: A column whose unique values will be used to create separate lines (e.g., `'category_0_name'`).
+- **`x_col`**: The column to use as the x-axis (e.g., `'months_since_competitor_opened'`). **Warning**: If a datetime
+  or datetime-like column is passed, a warning will suggest using the `time_plot` module instead.
+- **`title`**, **`x_label`**, **`y_label`**: Custom text for the plot title and axis labels.
+- **`legend_title`**: Custom title for the legend based on `group_col`.
+- **`move_legend_outside`**: Boolean flag to move the legend outside the plot.
+
+---
+
+### Dependencies:
+- `pandas`: For DataFrame manipulation and grouping.
+- `matplotlib`: For generating plots.
+- `pyretailscience.style.graph_utils`: For applying consistent graph styles across the plots.
+
+"""
+
+import logging
+
+import pandas as pd
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.graph_utils import GraphStyles
+from pyretailscience.style.tailwind import COLORS
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+def _check_datetime_column(df: pd.DataFrame, x_col: str) -> None:
+    """Checks if the x_col is a datetime or convertible to datetime.
+
+    Issues a warning if the column is datetime-like, recommending
+    the use of a time-based plot.
+
+    Args:
+        df (pd.DataFrame): The dataframe containing the column to check.
+        x_col (str): The column to check for datetime-like values.
+    """
+    if pd.api.types.is_datetime64_any_dtype(df[x_col]):
+        logging.warning(
```

Comment:
For now I'm not using logging anywhere and just using the standard python warnings. At least for now the idea is that logging is up to the user.